java/src/jmri/implementation/DefaultIdTag.java

```diff
@@ -44,6 +45,20 @@ public DefaultIdTag(String systemName, String userName) {
         setWhereLastSeen(null);
     }
 
+    @Override
+    public int compareTo(NamedBean n2) {
+        Objects.requireNonNull(n2);
+        String o1 = this.getSystemName();
+        String o2 = n2.getSystemName();
+        int p1len = Manager.getSystemPrefixLength(o1);
+        int p2len = Manager.getSystemPrefixLength(o2);
+        int comp = o2.substring(0, p1len).compareTo(o2.substring(0, p2len));
```

Comment:
Are you sure it should be `o2.substring(0, p1len)` and not `o1.substring(0, p1len)`?

I think this line should be:

`int comp = o1.substring(0, p1len).compareTo(o2.substring(0, p2len));`